x-pack/platform/test/onechat_functional/tests/converse/conversation_history.ts

```diff
@@ -0,0 +1,282 @@
+
+      const deleteConv1Button = await testSubjects.find('delete-conversation-button');
+      await deleteConv1Button.click();
+
+      // Confirm deletion in modal
+      const confirmDeleteButton1 = await testSubjects.find('confirmModalConfirmButton');
+      await confirmDeleteButton1.click();
+
+      // Wait for Conversation1 to be removed (deletion to complete) and conversation list to update
+      await retry.try(async () => {
+        const conversationList = await testSubjects.find('agentBuilderConversationList');
+        const conversationText = await conversationList.getVisibleText();
+        expect(conversationText).not.to.contain('Conversation1');
+        expect(conversationText).to.contain('Conversation2');
+        expect(conversationText).to.contain('Conversation3');
+      });
+    });
+
+    it('can continue chatting in an existing conversation', async () => {
+      const MOCKED_INPUT = 'hello conversation 2';
+      const MOCKED_RESPONSE = 'responding from conversation 2';
+
+      await common.navigateToApp(APP_ID, { path: 'conversations/new' });
+
+      // Wait for conversations to load and assert all 3 are present
+      await retry.try(async () => {
+        const conversationList = await testSubjects.find('agentBuilderConversationList');
+        const conversationText = await conversationList.getVisibleText();
+        expect(conversationText).to.contain('Conversation1');
+        expect(conversationText).to.contain('Conversation2');
+        expect(conversationText).to.contain('Conversation3');
+      });
+
+      // Select Conversation2
+      const conv2Button = await testSubjects.find('conversationItem-conv-2');
+      await conv2Button.click();
+
+      // Wait for Conversation2 to load and assert its content
+      await retry.try(async () => {
+        const titleElement = await testSubjects.find('agentBuilderConversationTitle');
+        const titleText = await titleElement.getVisibleText();
+        expect(titleText).to.contain('Conversation2');
+
+        const responseElement = await testSubjects.find('agentBuilderRoundResponse');
+        const responseText = await responseElement.getVisibleText();
+        expect(responseText).to.contain('This is the response for conversation 2');
+      });
+
+      // Set up LLM proxy to respond to the new message
+      void llmProxy.interceptors.userMessage({
+        when: ({ messages }) => {
+          const lastMessage = messages[messages.length - 1]?.content as string;
+          return lastMessage.includes(MOCKED_INPUT);
+        },
+        response: MOCKED_RESPONSE,
+      });
+
+      // Type and send the new message
+      const inputField = await testSubjects.find('onechatAppConversationInputFormTextArea');
+      await inputField.click();
+      await inputField.type(MOCKED_INPUT);
+
+      const submitTextButton = await testSubjects.find(
+        'onechatAppConversationInputFormSubmitButton'
+      );
+      await submitTextButton.click();
```

Comment:
Another example of what could be extracted to a helper function on the page object, e.g `submitMessage(msg: string)` or something